File: mpi-backend/mpi-backend/src/main/java/itmo/mpi/entity/TripRequest.java
package itmo.mpi.entity;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import java.time.LocalDate;

@Entity
@Table(name = "trip_request")
@Getter
@Setter
public class TripRequest {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "uid", nullable = false)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "traveler", nullable = false)
    private User traveler;

    @Column(name = "date_start")
    private LocalDate dateStart;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "island_start", nullable = false)
    private Island islandStart;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "island_end", nullable = false)
    private Island islandEnd;

    @Column(name = "status", length = 20)
    private String status;

}

File: mpi-frontend/src/app/entity/TripRequest.ts
import { Crew } from "./Crew";
import { Island } from "./island";
import { Ship } from "./Ship";
import { TripRequestStatus } from "./TripRequestStatus";
import { UserInfo } from "./UserInfo";

export class TripRequest {
    public id?: number;
    public traveler?: UserInfo;
    public dateStart: Date;
    public dateEnd: Date;
    public islandStart: Island;
    public islandEnd: Island;
    public status: TripRequestStatus;
    public ship: Ship;
    public crew: Crew;
    public cost: number;
    public isRated: boolean;
}
